### PR TITLE
Add Jonathan Tong to maintainers

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Jonathan has consistently demonstrated expertise and endurance on the CAPZ project, and we will be grateful to have him as a maintainer.

Here is a list of substantial PRs primarily reviewed by Jonathan:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1873
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1918
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2007
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2168
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1686
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3847
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3843
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3340
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3847
- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4413

Thank you, Jonathan!

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Jont828** [comment]
Thanks for the nomination @jackfrancis, it's been a pleasure to work with everyone!

**Comment by willie-yao** [comment]
/lgtm

Congrats Jonathan! You have done some amazing work for this project and this is well deserved 🚀

File: OWNERS_ALIASES [hunk 1]
```diff
@@ -18,6 +18,7 @@ aliases:
     - jackfrancis
     - mboersma
     - nojnhuh
+    - Jont828
   cluster-api-azure-reviewers:
     - Jont828
```
Comment (Jont828) [inline OWNERS_ALIASES:23]: Should we remove me from reviewers in that case?

Comment (jackfrancis) [reply OWNERS_ALIASES:23]: Good catch, that seems to be the pattern, I'll update

**Review by mboersma — APPROVED** [review]
/lgtm

@Jont828 has been productive and dedicated to CAPZ (and to the sig-cluster-lifecycle universe) for many moons. Congratulations and thank you for stepping up!

**Comment by nojnhuh** [comment]
+1, thank you @Jont828!
/approve

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by nawazkh** [comment]
Late to the party, but yay! Keep up the good work!
/lgtm

**Pull request merged**